04_numeronym/davidbroughsmyth/main_test.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := strconv.Quote("[a11y K8s abc]\n")
+	actual := strconv.Quote(buf.String())
+	assert.Equal(t, expected, actual, "Unexpected output from main()")
+}
+
+var tests = map[string]struct {
+	input, want []string
+}{
+	"empty":        {input: []string{}, want: []string{}},
+	"SpecialChars": {input: []string{"@@!##%&&"}, want: []string{"@6&"}},
+	"Alphanumeric": {input: []string{"8abcABC5Z"}, want: []string{"87Z"}},
+	"StarTrek":     {input: []string{"Beam", "me", "up", "Scotty"}, want: []string{"B2m", "me", "up", "S4y"}},
+	"MaryPoppins": {
+		input: []string{"It's", "supercalifragilisticexpialidocious"},
+		want:  []string{"I2s", "s32s"},
+	},
+}
```

Comment:
You need to add some test cases that start and/or end with multi-byte runes, as well as including strings containing multi-byte runes. Make sure you have three runes where some are multi-byte. By multi-byte I mean what people refer to as unicode characters - anything that is utf-8 encoded as multiple bytes such as emojis.

Have a read of https://blog.golang.org/strings for the gory details of strings.
I think you'll find some of your test case start failing in those cases.

Reply:
Oh, I now see that now.

Reply:
Thanks, the link gave me the ideas to fix this function.
I have refactored this solution to handle Unicode. And provide test cases for the multi-byte runes.